api/v1alpha2/conversion.go

```diff
@@ -111,6 +111,9 @@ func restoreMachineSpec(restored *v1alpha3.MachineSpec, dst *v1alpha3.MachineSpe
 		dst.ClusterName = restored.ClusterName
 	}
 	dst.Bootstrap.DataSecretName = restored.Bootstrap.DataSecretName
+	if restored.FailureDomain != nil {
+		dst.FailureDomain = restored.FailureDomain
+	}
```

Comment:
This can probably just be `dst.FailureDomain = restored.FailureDomain`

Reply:
ack, fixed